### DISCUSSION TITLE
DDR Character slain by sword

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#219] [Code Algorithm Questions] [answered]
Hi Chris,

Whilst doing a few comparisons with original DDR, I came across the following text where Luxor got killed by Carorthane who was carrying a sword object.

![DDR1](https://github.com/ChilliHugger/The-Lords-Of-Midnight/assets/91921640/76a69969-05c6-4794-b048-463bf7eacc0b)

I don't see anywhere in the codebase where the "killedbyobject" property is set as a result of night battles (only when fighting critters).

Might be worth a cross check

Regards
Andrew

**Comment by IcemarkUK** [comment, marked as answer]
I have to admit to never having read that text like that!

You are right, this is not how I describe it.

Checking the death description code...

https://github.com/ChilliHugger/The-Lords-Of-Midnight/blob/06479089b51e21ff86ffe805e3d5d5a45c5744d0/main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp#L331C1-L343C2

You can see that I'm using **fighting_against** to use the following text

```
SS_KILLED_BY
killed in battle by {char:battle:fighting:longname}.
```

which should should be set in the battle processor. However it also looks incorrect, as it seems to be set for the attacker not the defender. So if the lord dies and doesn't get an attacking phase, the info will not be correct.

The text should probably be

```
SS_KILLED_BY
killed in battle by {char:battle:fighting:object:name}.
```

**Reply by FrozenEmpire** [reply]
At least I'm assuming that it was referring to the object and wasn't just a static text to indicate he got killed in battle by sword... could be either way actually :)

**Reply by IcemarkUK** [reply]
Original code **is** picking object type.

**Reply by IcemarkUK** [reply]
I was idly thinking about this and something was bothering me...

So looking at this a little further...

Here is the original code. So the characters object has been changed to the one that kills them. 0-127 will be lords and 128+ will be objects.

If it's an object then 5 is subtracted and this is added to the token buffer. So object 128 will be token 123.

Token 123 is 'Dragons' which makes sense - BUT only if fighting a nasty. Therefore as the comment says - this is not an object but a nasty. BUT that doesn't make any sense as far as the text you got. It was quite clearly 'sword'

```
StateACharacterIsDead:                                                ;L7E60
       CALL   FullCharTitle
       CALL   Bytes_Print_Buffer
       DB     0034h,00C0h,0FEh,02Ch,09Eh,0BAh,0FFh             ; 'Is Dead, Slain By'
       CALL   GetCharObject
       CMP    AL,080h                                          ;Greater than #80
       JNC    L7E80                                            ;is a nasty and not
       MOV    BYTE PTR CharacterNumber,AL                      ;a person
       CALL   FullCharTitle
L7E7A: CALL   StopToBuffer
       JMP    FlushPrintBuffer
L7E80: SUB    AL,005h
       CALL   A_IntoPrintBuffer
       JMP    L7E7A
```


So dug deeper...

There is
``` 
       MOV    AL,BYTE PTR 099h
       MOV    BYTE PTR CharFightsWhat,AL
       CALL   WhatAreFightConsequences
```

```
       MOV    AL,BYTE PTR 099h
       MOV    BYTE PTR CharFightsWhat,AL
       CALL   GetArmy
```

```
       MOV    AL,BYTE PTR WhatsAtLocation
       ADD    AL,BYTE PTR 080h
       MOV    BYTE PTR CharFightsWhat,AL
       JMP    WhatAreFightConsequences
```

```
L7F7C: MOV    AL,BYTE PTR CharacterNumber
       MOV    BYTE PTR CharFightsWhat,AL
```

And more importantly

```
CharacterDies:                                                        ;L8021
       XOR    AL,AL
       CALL   PutCharReckless
       CALL   GetCharObject                                    ;
       CMP    AL,Nothing                                       ;is he Carrying Anything?
       JZ     L803B                                            ;
       MOV    BYTE PTR ObjectNumber,AL
       CALL   GetCharLocation_1                                ;
       CALL   PutObjectLocation_1                              ;Drop The Object at
       CALL   GetCharLocation_2                                ;the current location
       CALL   PutObjectLocation_2                              ;
L803B: MOV    AL,BYTE PTR CharFightsWhat
       CALL   PutCharObject                                    ;Did he fight a nasty
       CMP    AL,080h                                          ;or a Character?
       JC     @F                                               ;Return if Nasty
       RET
```

So before a fight of any type, CharFightsWhat is set to either 0x99, the character number, or 0x80+nasty. Nasties being
0 - Dragons
1 - Wolves
2 - Trolls
3 - Skulkrin

If you check the tokens out, token (0x99 - 5 = 0x94) is 'sword'

So that description text will either read

Slain by sword ( ie: killed by army )
Slain by [character full name]
Slain by [nasty]

**Reply by FrozenEmpire** [reply]
Thanks for the comprehensive investigation... I also had those 2nd thoughts that it might just be standard battle terminology when fighting with armies.

However I do quite like knowing that they died at the hands of one of the unique weapons :)

There's a dilemma...

**Reply by IcemarkUK** [reply]
tbf... I was thinking of adding that additional option in "slain by the spear of...." etc or "slain by bigman the giant using the hammer of destiny"

**Reply by FrozenEmpire** [reply]
Yeah like that - could well be how it ends up for me :)
